core/web/pipeline_runs_controller.go

```diff
@@ -84,30 +85,49 @@ func (prc *PipelineRunsController) Create(c *gin.Context) {
 	}
 	idStr := c.Param("ID")
 
+	user, isUser := authenticatedUser(c)
+	ei, _ := authenticatedEI(c)
+	authorizer := webhook.NewAuthorizer(prc.App.GetStore().DB, user, ei)
+
 	// Is it a UUID? Then process it as a webhook job
 	jobUUID, err := uuid.FromString(idStr)
 	if err == nil {
-		jobRunID, err2 := prc.App.RunWebhookJobV2(context.Background(), jobUUID, string(bodyBytes), pipeline.JSONSerializable{Null: true})
+		canRun, err2 := authorizer.CanRun(jobUUID)
```

Comment:
I think we want CanRun to accept this request context and pass it to the db query to respect its cancellation (e.g. https://github.com/smartcontractkit/chainlink/blob/9e4afecb74e5bfb05a982a5b090a7c781ec84fba/core/web/jobs_controller.go#L125)

Reply:
Great idea